llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/SeedCollector.h

```diff
@@ -0,0 +1,136 @@
+//===- SeedCollector.h ------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This file contains the mechanism for collecting the seed instructions that
+// are used as starting points for forming the vectorization graph.
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_TRANSFORMS_VECTORIZE_SANDBOXVECTORIZER_SEEDCOLLECTOR_H
+#define LLVM_TRANSFORMS_VECTORIZE_SANDBOXVECTORIZER_SEEDCOLLECTOR_H
+
+#include "llvm/ADT/BitVector.h"
+#include "llvm/ADT/iterator_range.h"
+#include "llvm/Analysis/ScalarEvolution.h"
+#include "llvm/SandboxIR/Instruction.h"
+#include "llvm/SandboxIR/Utils.h"
+#include "llvm/SandboxIR/Value.h"
+#include <iterator>
+#include <memory>
+
+namespace llvm::sandboxir {
+class Instruction;
+class StoreInst;
+class BasicBlock;
+
+/// An ordered set of Instructions that can be vectorized.
+class SeedBundle {
+public:
+  using SeedList = SmallVector<sandboxir::Instruction *>;
```

Comment:
The SeedList type is used for the internal vector but it's also used in the constructor, which I find a bit confusing. 
Since it's used internally it should not be public, because this is exposing an implementation detail, but since it's also used in the constructor then it should be public? 

If we make this private, would there be a real need to have a separate typedef for the type used in the constructor ? I don't think we would really need one.

Reply:
The type is also returned as a MutableArrayReference by getSlice. So the question here would be three-fold:

1. How do we expect clients to iterate over slices?
2. Would they need to construct new SeedBundles out of slices?
3. Is it more convenient to do either of the above with a type?

I'm not particularly attached, but once I find myself using the same type in multiple places, I like a name for it, personally.